### PR TITLE
Feature/failure attempts

### PR DESCRIPTION
**Pull request by AniJan**

### CODE REVIEW

File: identity-api/identity-api-facade/src/main/java/com/sflpro/identity/api/endpoints/AuthenticationEndpoint.java [hunk 1]
```diff
@@ -75,18 +84,24 @@
...
             logger.info("Done authenticating by credential type:'{}'.", requestDto.getDetails().getCredentialType());
             return mapper.map(authResponse, AuthenticationResponseDto.class);
         } catch (InactiveIdentityException e) {
             logger.warn("Authentication failed for inactive identity request:'{}'.", requestDto);
             throw new InactiveIdentityExceptionDtoDto(e);
         } catch (AuthenticationServiceException e) {
             logger.warn("Authentication failed for request:'{}'.", requestDto);
+            final Credential credential = authService.getCredential(authRequest);
+            credential.setFailedAttempts(credential.getFailedAttempts() + 1);
+            Credential update = credentialService.update(credential);
```
Comment (sfl-sonar) [inline identity-api/identity-api-facade/src/main/java/com/sflpro/identity/api/endpoints/AuthenticationEndpoint.java]: MAJOR Remove this useless assignment to local variable "update". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
MINOR Remove this unused "update" local variable. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)

File: identity-api/identity-api-facade/src/main/java/com/sflpro/identity/api/endpoints/AuthenticationEndpoint.java [hunk 2]
```diff
@@ -75,18 +84,24 @@
...
-                    permissions.add(permission.getName())
-                )
+                    role.getPermissions().forEach(permission ->
+                            permissions.add(permission.getName())
+                    )
             );
             authResponse.setPermissions(permissions);
+            final Credential credential = authService.getCredential(authRequest);
+            credential.setFailedAttempts(0);
+            Credential update = credentialService.update(credential);
```
Comment (sfl-sonar) [inline identity-api/identity-api-facade/src/main/java/com/sflpro/identity/api/endpoints/AuthenticationEndpoint.java]: MAJOR Remove this useless assignment to local variable "update". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
MINOR Remove this unused "update" local variable. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)

**Comment by sfl-sonar** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: identity-api/identity-api-facade/src/main/java/com/sflpro/identity/api/endpoints/AuthenticationEndpoint.java [hunk 3]
```diff
@@ -59,6 +58,15 @@
     @Autowired
     private AuthenticationService authService;
 
+    @Autowired
+    CredentialService credentialService;
```
Comment (mikron) [inline identity-api/identity-api-facade/src/main/java/com/sflpro/identity/api/endpoints/AuthenticationEndpoint.java]: Don't need this services

File: identity-core/identity-core-services-impl/src/main/java/com/sflpro/identity/core/services/auth/impl/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -81,4 +111,11 @@ public AuthenticationServiceImpl(AuthenticatorRegistry authenticatorRegistry, Pr
     public void invalidateToken(TokenInvalidationRequest tokenRequest) throws TokenServiceException {
         tokenService.invalidateToken(tokenRequest);
     }
+
+    @Override
+    public <T extends Credential, E extends CredentialIdentifier<T>, S extends AuthenticationRequestDetails<T, E>> T getCredential(AuthenticationRequest<T, E, S> request) {
```
Comment (mikron) [inline identity-core/identity-core-services-impl/src/main/java/com/sflpro/identity/core/services/auth/impl/AuthenticationServiceImpl.java]: Don't need this method here, change access to private or remove it

File: identity-api/identity-api-facade/src/main/java/com/sflpro/identity/api/endpoints/AuthenticationEndpoint.java [hunk 5]
```diff
@@ -8,12 +8,14 @@
 import com.sflpro.identity.api.common.dtos.identity.InactiveIdentityExceptionDtoDto;
 import com.sflpro.identity.api.common.dtos.token.TokenInvalidationRequestDto;
 import com.sflpro.identity.api.mapper.BeanMapper;
-import com.sflpro.identity.core.db.entities.Role;
 import com.sflpro.identity.core.services.auth.AuthenticationRequest;
 import com.sflpro.identity.core.services.auth.AuthenticationResponse;
 import com.sflpro.identity.core.services.auth.AuthenticationService;
 import com.sflpro.identity.core.services.auth.AuthenticationServiceException;
+import com.sflpro.identity.core.services.credential.CredentialService;
```
Comment (sfl-sonar) [inline identity-api/identity-api-facade/src/main/java/com/sflpro/identity/api/endpoints/AuthenticationEndpoint.java]: MINOR Remove this unused import 'com.sflpro.identity.core.services.credential.CredentialService'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by sfl-sonar** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**